mon-pix/app/components/certification-starter.js

```diff
@@ -2,47 +2,46 @@ import { inject as service } from '@ember/service';
 import Component from '@ember/component';
 
 export default Component.extend({
```

Comment:
De manière générale tout le contenu ici devrait être au niveau controller, les actions de store et de router se font à ce niveau là et sont passées comme arguments au component enfant.

Reply:
Je suis pas sûr de voir l'intérêt. Tout ce que je lis sur Ember depuis quelque temps c'est que les `controller` sont un concept legacy qui va sans doute un jour sauter, en faveur d'une archi complètement orientée composant/services. Y'a qu'à voir la documentation qui n'a pas changée depuis 10 versions où ils disent de les utiliser juste pour gérer les query params. Ils ont fail de les supprimé par le passé avec les {{routable-components}} et en ont tiré comme lesson de pas faire des gros majeur re-design upfront, mais c'est pas pour autant que je parierai sur ce type d'objet surtout quand tout peut tenir sur des composants avec des patterns qu'on retrouve dans d'autres framework.
Si tu as un exemple concret de pourquoi c'est mieux de faire comme tu le suggères, je suis preneur en revanche

Reply:
Plusieurs raisons :
- par cohérence, on fait comme ça dans le reste du code ça permet de s'y retrouver
- je sais pas ce qu'Ember veut faire de ses controllers mais quoiqu'il en soit la notion de routing n'a pas lieu d'être dans un component, il est censé être théoriquement indépendant et combinatoire
- sur la logique de store si on suit le pattern data-down, action-up (bien qu'il soit un peu twisté en Ember) cette logique est défini au plus haut niveau (si ce n'est pas dans un controller c'est dans un service)

Reply:
Pour moi le pattern `DDAU` c'est la réponse d'Ember à `Redux` qui a mis KO le `two-way` binding. 

Le composant parent passe des données, lesquels ne peuvent pas modifier les données du composant parent, _sauf_ dans une action définit _sur le composant parent_ (et ça fait un cercle, 1 way binding, comme redux). La seule exception en Ember à ça, c'est le helper `{{input}}` et le helper `{{ (mut ... ) }}` Cf. la réponse de Edward Faulker un core committer sur ce thread : https://discuss.emberjs.com/t/rationale-for-data-down-actions-up/13830/3 

Qu'un composant soit `smart` à savoir contienne de la logique **métier** = il sait où transitionner dans l'appli, ça me semble tout à fait normal. EmberMap en parle dans la série sur le component design que je vous avais forward sur slack.